### ISSUE TITLE
Track symbol table in python

### ISSUE THREAD

**Issue by reuster986** [Bears-R-Us/arkouda#98]
Store a module-level dictionary that tracks pdarrays
* key: python name of array
* val: (server-side name, size, dtype, ...)

User may have to call a function to scrape the python names.

**Comment by mhmerrill** [comment]
linked to issue #18

**Comment by milthorpe** [comment]
I believe this enhancement would fix a (minor) bug I observed after shutting down and restarting the arkouda server. I realize it's probably not an important use case, but anyway:

1. Startup local arkouda server
2. Client:
```
>>> import arkouda as ak
>>> ak.connect(connect_url='tcp://localhost:5555')
connected to arkouda server tcp://*:5555
>>> A = ak.arange(10)
>>> print(A)
[0 1 2 3 4 5 6 7 8 9]
>>> ak.shutdown()
```
3. Startup local arkouda server (again, so it's a different server)
4. Client:
```
>>> ak.connect(connect_url='tcp://localhost:5555')
>>> print(A)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/noback/milthorpe/arkouda/arkouda/pdarrayclass.py", line 134, in __str__
    return generic_msg(cmd='str', args='{} {}'.format(self.name,pdarrayIterThresh))
  File "/noback/milthorpe/arkouda/arkouda/client.py", line 548, in generic_msg
    return _send_string_message(cmd=cmd, args=args, recv_binary=recv_binary)
  File "/noback/milthorpe/arkouda/arkouda/client.py", line 351, in _send_string_message
    raise RuntimeError(return_message.msg)
RuntimeError: UnknownSymbolError Line 236 In MultiTypeSymbolTable.checkTable: Error: datastr: unknown symbol: id_mHHKkRm_2
```

The Python array `A` is not deleted during the call to `arkouda.shutdown()`, and afterwards holds a reference (name) to a non-existent server-side array.

**Comment by Ethan-DeBandi99** [comment]
@milthorpe - The functionality I believe you are getting at here is only supported when using `ak.disconnect()` because the server persists. You can achieve what you are trying to do by writing out your variables to a file and loading them on the new server from the file. Currently, HDF5 and Parquet file formats are supported.

**Comment by Ethan-DeBandi99** [comment]
@pierce314159 reviewed this and believe it can be closed because it duplicates #18.

**Comment by milthorpe** [comment]
I'm not expecting the array `A` to continue to behave correctly after calling `ak.shutdown()`; as it is stored on the server, it makes sense that it disappears. (On the other hand, calling `ak.disconnect()` followed by `ak.connect()` to reconnect to the same server seems like a reasonable use case, and we might expect arkouda arrays to continue to behave correctly afterwards.)
What I think is buggy here is that after calling `ak.shutdown()`,  we have a local array `A` that refers to a non-existent server-side array, with the result that no operation on `A` can succeed. 
A small improvement might be for arkouda to note that a particular local array object no longer has a server-side array, and throw an exception to indicate this. I figured that one way to implement this was suggested by this issue, as keeping a dictionary of the local pdarrays would allow arkouda to remove the local reference to the server when `ak.shutdown()` is called.
